R/processing_pcs.R

```diff
@@ -61,3 +61,68 @@ downsample_pcs <- function(pcs, frac_cells, min_cells = 50) {
 
   return(downsampled_pcs)
 }
+
+#' Calculate within-batch ARI using provided PCs for use in integration metric calculations
+#'
+#' @param merged_sce The merged SCE object containing data from multiple batches
+#' @param pc_name The name used to access the PCA results stored in the
+#'   SingleCellExperiment object; default is "PCA"
+#' @param batch_column The variable in `merged_sce` indicating the grouping of interest.
+#'  Generally this is either batches or cell types. Default is "library_id".
+#'
+#' @return Tibble with three columns: `ari`, representing the calculated ARI values;
+#'   `batch_id`, the batch id associated with each `ari`; `pc_name`, the name
+#'   associated with the pc results
+
+calculate_within_batch_ari_pcs <-
+  function(merged_sce, pc_name, batch_column = "library_id") {
+    # Pull out the PCs or analogous reduction from merged object
```

Comment:
For good practice, we should add the check for `pc_name` being in the `merged_sce` and same with the `batch_column`.